core/src/main/java/org/elasticsearch/common/lucene/Lucene.java

```diff
@@ -236,13 +236,8 @@ public static void checkSegmentInfoIntegrity(final Directory directory) throws I
             protected Object doBody(String segmentFileName) throws IOException {
                 try (IndexInput input = directory.openInput(segmentFileName, IOContext.READ)) {
                     final int format = input.readInt();
-                    final int actualFormat;
                     if (format == CodecUtil.CODEC_MAGIC) {
```

Comment:
I saw the leniency too. but can we break this out? I don't think these kinds of changes belong in a lucene upgrade!